frontend/src/metabase/visualizations/lib/renderer_utils.js

```diff
@@ -346,10 +345,14 @@ export function xValueForWaterfallTotal({ settings, series }) {
     const lastXValue = xValues[xValues.length - 1];
     return lastXValue.clone().add(count, interval);
   } else if (isQuantitative(settings) || isHistogram(settings)) {
-    return xValues[xValues.length - 1] + xInterval;
+    const lastXValue = xValues[xValues.length - 1];
+    return lastXValue + xInterval;
+  } else if (isOrdinal(settings)) {
+    const maxXValue = xValues.reduce((a, b) => (a > b ? a : b), xValues[0]);
```

Comment:
What do you think about using `Math.max(...xValues)` for a bit better readability?


Also, maybe returning `Infinity` here would be fine even though it is hackish. Infinity value can't come from the back end, in a super rare case with `maxXValue === Number.MAX_SAFE_INTEGER` it will still work correctly. And it can be readable enough if we define it as a constant `frontend/src/metabase/lib/constants.js` with an explanation similarly to `NULL_NUMERIC_VALUE`